rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java

```diff
@@ -81,8 +80,15 @@ private void fillUnresolvablePoms() {
     @Override
     public CacheResult<RawMaven> computeMaven(URI repo, String groupId, String artifactId, String version,
                                               Callable<RawMaven> orElseGet) throws Exception {
-        // FIXME key be repo as well, because different repos may have different versions of the same POM
-        String cacheKey = groupId + ':' + artifactId + ':' + version;
+
+        //There are a few exceptional artifacts that will never be resolved by the repositories. This will always
+        //result in an Unavailable response from the cache.
+        String artifactCoordinates = groupId + ':' + artifactId + ':' + version;
+        if (unresolvablePoms.contains(artifactCoordinates)) {
+            return UNAVAILABLE_POM;
+        }
+
+        String cacheKey = repo.toString() + ":" + artifactCoordinates;
```

Comment:
Note, I first create just the group/artifact/version key and check it against the unresolvable list. I return UNAVAILABEL_POM immediately if we encounter one of these poms. 

Then I add the repository to the key and check the cache. The cache will have an Optional.empty() if it is not found in the repository (and that also results in an UNAVAILABLE_POM.